library/ssl_tls13_server.c

```diff
@@ -2628,7 +2646,13 @@ static int ssl_tls13_handshake_wrapup( mbedtls_ssl_context *ssl )
     mbedtls_ssl_tls13_handshake_wrapup( ssl );
 
 #if defined(MBEDTLS_SSL_SESSION_TICKETS)
-    mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_TLS1_3_NEW_SESSION_TICKET );
+#if defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_PSK_ENABLED)
+    /* Sent NewSessionTicket message only when client supports PSK */
+    if( ! mbedtls_ssl_tls13_some_psk_enabled( ssl ) )
+        mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_HANDSHAKE_OVER );
+    else
+#endif
```

Comment:
```suggestion
    /* Sent NewSessionTicket message only when client supports PSK */
    if( ! mbedtls_ssl_tls13_some_psk_enabled( ssl ) )
        mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_HANDSHAKE_OVER );
    else
```

SESSION_TICKETS depends on SOME_PSK_ENABLED. If current code does not match that, it is a bug.

Reply:
I know that, but this would cause CI failure. Any suggestion?

Reply:
I have add comments showing this is a know issue and it is expected to be fixed with #6395.